runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -6482,6 +6482,51 @@ TR::Register *J9::Power::TreeEvaluator::VMnewEvaluator(TR::Node *node, TR::CodeG
             genInitArrayHeader(node, iCursor, isVariableLen, clazz, NULL, resReg, zeroReg, condReg, enumReg, dataSizeReg,
                   tmp5Reg, tmp4Reg, conditions, needZeroInit, cg);
 
+#ifdef TR_TARGET_64BIT
+         TR::Register *offsetReg = tmp5Reg;
+         TR::Register *firstDataElementReg = tmp4Reg;
+         TR::MemoryReference *dataAddrSlotMR = NULL;
+
+         if (TR::Compiler->om.compressObjectReferences() && isVariableLen)
+            { // We need to check sizeReg at runtime to determine correct offset of dataAddr field.
+            if (comp->getOption(TR_TraceCG))
+               traceMsg(comp, "Dealing with compressed refs variable length array.\n");
+            TR_ASSERT_FATAL_WITH_NODE(node, (fej9->getOffsetOfDiscontiguousDataAddrField() - fej9->getOffsetOfContiguousDataAddrField()) == 8, "DataAddr field offset for discontiguous arrays is expected to be 8 bytes more than dataAddr field offset for contiguous arrays.\n");
```

Comment:
This assert is ok, but what would be better is to print the offset of discontiguous and contiguous field in the assert. Similarly for the other fatal assert.